src/storage/storage.cc

```diff
@@ -75,12 +76,46 @@ const int64_t kIORateLimitMaxMb = 1024000;
 
 using rocksdb::Slice;
 
+static Status CreateSnapshot(Config &config, const std::string &snapshot_location) {
```

Comment:
Actually I don't really understand how this works, seems this create a checkpoint when opening the file? Do you need create it on server and consume it on cli?

Reply:
> Do you need create it on server and consume it on cli?
I'm not sure i understand the second part of your question
> Actually I don't really understand how this works, seems this create a checkpoint when opening the file?
As for how this works, it will read the data from the `dir` passed by the user, then use RocksDB snapshot functionality to make a read-only copy of that data into the `snapshot-dir` directory, then spawn a read-only server on that copy.

This is helpful for my team to debug a running kvrocks app without accidentally changing the values of the data or closing the existing read-write server.